code/modules/mob/living/simple_animal/hostile/swarmer.dm

```diff
@@ -0,0 +1,834 @@
+	hud_possible = list(ANTAG_HUD, DIAG_STAT_HUD, DIAG_HUD)
+	obj_damage = 0
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	attack_verb_continuous = "shocks"
+	attack_verb_simple = "shock"
+	attack_sound = 'sound/effects/empulse.ogg'
+	friendly_verb_continuous = "pinches"
+	friendly_verb_simple = "pinch"
+	speed = 0
+	faction = list("swarmer")
+	AIStatus = AI_OFF
+	pass_flags = PASSTABLE
+	mob_size = MOB_SIZE_TINY
+	ventcrawler = VENTCRAWLER_ALWAYS
+	ranged = 1
+	projectiletype = /obj/projectile/beam/disabler/swarmer
+	ranged_cooldown_time = 20
+	projectilesound = 'sound/weapons/taser2.ogg'
+	loot = list(/obj/effect/decal/cleanable/robot_debris, /obj/item/stack/ore/bluespace_crystal)
+	del_on_death = 1
+	deathmessage = "explodes with a sharp pop!"
+	light_color = LIGHT_COLOR_CYAN
+	hud_type = /datum/hud/swarmer
+	speech_span = SPAN_ROBOT
+	///Resource points, generated by consuming metal/glass
+	var/resources = 0
+	///Maximum amount of resources a swarmer can store
+	var/max_resources = 100
+	///List used for player swarmers to keep track of their drones
+	var/list/mob/living/simple_animal/hostile/swarmer/melee/dronelist = list()
+
+/mob/living/simple_animal/hostile/swarmer/Initialize()
+	. = ..()
+	verbs -= /mob/living/verb/pulled
+	for(var/datum/atom_hud/data/diagnostic/diag_hud in GLOB.huds)
+		diag_hud.add_to_hud(src)
+
+/mob/living/simple_animal/hostile/swarmer/med_hud_set_health()
+	var/image/holder = hud_list[DIAG_HUD]
+	var/icon/I = icon(icon, icon_state, dir)
+	holder.pixel_y = I.Height() - world.icon_size
+	holder.icon_state = "huddiag[RoundDiagBar(health/maxHealth)]"
+
+/mob/living/simple_animal/hostile/swarmer/med_hud_set_status()
+	var/image/holder = hud_list[DIAG_STAT_HUD]
+	var/icon/I = icon(icon, icon_state, dir)
+	holder.pixel_y = I.Height() - world.icon_size
+	holder.icon_state = "hudstat"
+
+/mob/living/simple_animal/hostile/swarmer/Stat()
+	..()
+	if(statpanel("Status"))
+		stat("Resources:",resources)
+
+/mob/living/simple_animal/hostile/swarmer/emp_act()
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	if(health > 1)
+		adjustHealth(health-1)
+	else
+		death()
+
+/mob/living/simple_animal/hostile/swarmer/CanAllowThrough(atom/movable/O)
+	. = ..()
+	if(istype(O, /obj/projectile/beam/disabler))//Allows for swarmers to fight as a group without wasting their shots hitting each other
+		return TRUE
+	if(isswarmer(O))
+		return TRUE
+
+////CTRL CLICK FOR SWARMERS AND SWARMER_ACT()'S////
+/mob/living/simple_animal/hostile/swarmer/AttackingTarget()
+	if(!isliving(target))
+		return target.swarmer_act(src)
+	else if(iscyborg(target))
+		var/mob/living/silicon/borg = target
+		borg.adjustBruteLoss(melee_damage_lower)
+		return ..()
+	else
+		return ..()
+		
+/mob/living/simple_animal/hostile/swarmer/MiddleClickOn(atom/A)
+	. = ..()
+	if(dronelist.len == 0)
+		return
+	var/turf/T = get_turf(A)
+	if(!T)
+		return
+	for(var/mob/living/simple_animal/hostile/swarmer/melee/drone in dronelist)
```

Comment:
Will there be anything besides drones in this list?
If not, avoid type-checking.